terraform/ecs_asg/iam.tf

```diff
@@ -0,0 +1,23 @@
+resource "aws_iam_role_policy" "publish_to_sns" {
+  name   = "publish_to_sns"
+  role   = "${aws_iam_role.role.id}"
+  policy = "${var.publish_to_sns_policy}"
+}
```

Comment:
Where is this resource used?

Separately, giving it a fixed name may cause issues – names are in a global namespace, I think, so you’d end up blatting over policy from a different instance of the module.